File: database/src/main/java/dreammaker/android/expensetracker/database/dao/AnalyticsDao.java
package dreammaker.android.expensetracker.database.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Query;
import dreammaker.android.expensetracker.database.view.AssetLiabilitySummary;

@Dao
@SuppressWarnings("unused")
public interface AnalyticsDao {

    @Query("SELECT * FROM `asset_liability_summary_view`")
    LiveData<AssetLiabilitySummary> getTotalAssetLiability();
}

File: database/src/main/java/dreammaker/android/expensetracker/database/dao/AnalyticsDao.kt
package dreammaker.android.expensetracker.database.dao

import androidx.room.Dao
import androidx.room.Query
import kotlinx.coroutines.flow.Flow

@Dao
interface AnalyticsDao {

    @Query("SELECT COALESCE(SUM(`balance`), 0) AS `total_balance` FROM `accounts`")
    fun getTotalAccountBalance(): Flow<Double>
}